src/tools/pkcs15-tool.c

```diff
@@ -1007,20 +1007,25 @@ static int read_ssh_key(void)
 		 * 2 x 4B item length headers
 		 * max 11B algorithm name, 32B key data */
 		unsigned char buf[64];
-		unsigned int len, n;
+		unsigned int n;
+		int len;
 
 		n = pubkey->u.eddsa.pubkey.len;
 		if (n != 32) {
 			fprintf(stderr, "Wrong public key length\n");
 			goto fail2;
 		}
 
+		len = snprintf(alg, sizeof alg, "ssh-ed25519");
```

Comment:
in this case, the `alg` could be a static string (unlike in the ecdsa case) to simplify things even more (the same case for RSA and DSA), but I do not insist on this change.

Reply:
done